Not compatible with Maven Enforcer Plugin 3.0.0?

It seems these rules don't work with the Maven Enforcer Plugin 3.0.0. I have a [reproducer project](https://github.com/mthmulders/mep-oss-issue) available. When I build it using Maven 3.8.2 with `mvn verify -e`, this stacktrace appears:

```console
Caused by: java.lang.NoSuchMethodError: 'org.apache.maven.shared.dependency.graph.DependencyNode org.apache.maven.shared.dependency.graph.DependencyGraphBuilder.buildDependencyGraph(org.apache.maven.project.MavenProject, org.apache.maven.artifact.resolver.filter.ArtifactFilter)'
    at org.sonatype.ossindex.maven.enforcer.BanVulnerableDependencies$Task.resolveDependencies (BanVulnerableDependencies.java:315)
    at org.sonatype.ossindex.maven.enforcer.BanVulnerableDependencies$Task.run (BanVulnerableDependencies.java:190)
    at org.sonatype.ossindex.maven.enforcer.BanVulnerableDependencies.execute (BanVulnerableDependencies.java:142)
    at org.apache.maven.plugins.enforcer.EnforceMojo.execute (EnforceMojo.java:200)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:78)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

Seem to have a related (or same issue).

```
[INFO] Maven Version: 3.8.4
[INFO] JDK Version: 17.0.1 normalized as: 17.0.1
[INFO] Java Vendor: Eclipse Adoptium
[INFO] OS Info: Arch: amd64 Family: unix Name: linux Version: 5.11.0-43-generic
```

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce (vulnerability-checks) on project xxx: Execution vulnerability-checks of goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce: java.lang.NoSuchMethodError: 'org.apache.maven.shared.dependency.graph.DependencyNode org.apache.maven.shared.dependency.graph.DependencyGraphBuilder.buildDependencyGraph(org.apache.maven.project.MavenProject, org.apache.maven.artifact.resolver.filter.ArtifactFilter)'
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.apache.maven.plugins:maven-enforcer-plugin:3.0.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/msjimlau/.m2/repository/org/apache/maven/plugins/maven-enforcer-plugin/3.0.0/maven-enforcer-plugin-3.0.0.jar
[ERROR] urls[1] = file:/home/msjimlau/.m2/repository/org/sonatype/ossindex/maven/ossindex-maven-enforcer-rules/3.1.0/ossindex-maven-enforcer-rules-3.1.0.jar
[ERROR] urls[2] = file:/home/msjimlau/.m2/repository/org/sonatype/ossindex/maven/ossindex-maven-common/3.1.0/ossindex-maven-common-3.1.0.jar
[ERROR] urls[3] = file:/home/msjimlau/.m2/repository/org/sonatype/ossindex/ossindex-service-client/1.1.1/ossindex-service-client-1.1.1.jar
[ERROR] urls[4] = file:/home/msjimlau/.m2/repository/org/sonatype/ossindex/ossindex-service-api/1.1.1/ossindex-service-api-1.1.1.jar
[ERROR] urls[5] = file:/home/msjimlau/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.9.0/jackson-annotations-2.9.0.jar
[ERROR] urls[6] = file:/home/msjimlau/.m2/repository/javax/ws/rs/javax.ws.rs-api/2.0.1/javax.ws.rs-api-2.0.1.jar
[ERROR] urls[7] = file:/home/msjimlau/.m2/repository/org/sonatype/goodies/package-url-java/1.0.1/package-url-java-1.0.1.jar
[ERROR] urls[8] = file:/home/msjimlau/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.15/jcl-over-slf4j-1.7.15.jar
[ERROR] urls[9] = file:/home/msjimlau/.m2/repository/joda-time/joda-time/2.9.9/joda-time-2.9.9.jar
[ERROR] urls[10] = file:/home/msjimlau/.m2/repository/com/google/code/gson/gson/2.8.2/gson-2.8.2.jar
[ERROR] urls[11] = file:/home/msjimlau/.m2/repository/org/apache/httpcomponents/httpclient/4.5.5/httpclient-4.5.5.jar
[ERROR] urls[12] = file:/home/msjimlau/.m2/repository/org/apache/httpcomponents/httpcore/4.4.9/httpcore-4.4.9.jar
[ERROR] urls[13] = file:/home/msjimlau/.m2/repository/com/google/guava/guava/28.1-android/guava-28.1-android.jar
[ERROR] urls[14] = file:/home/msjimlau/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
[ERROR] urls[15] = file:/home/msjimlau/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
[ERROR] urls[16] = file:/home/msjimlau/.m2/repository/org/checkerframework/checker-compat-qual/2.5.5/checker-compat-qual-2.5.5.jar
[ERROR] urls[17] = file:/home/msjimlau/.m2/repository/com/google/errorprone/error_prone_annotations/2.3.2/error_prone_annotations-2.3.2.jar
[ERROR] urls[18] = file:/home/msjimlau/.m2/repository/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar
[ERROR] urls[19] = file:/home/msjimlau/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar
[ERROR] urls[20] = file:/home/msjimlau/.m2/repository/org/apache/maven/shared/maven-dependency-tree/3.0.1/maven-dependency-tree-3.0.1.jar
[ERROR] urls[21] = file:/home/msjimlau/.m2/repository/org/codehaus/plexus/plexus-component-annotations/2.0.0/plexus-component-annotations-2.0.0.jar
[ERROR] urls[22] = file:/home/msjimlau/.m2/repository/org/eclipse/aether/aether-util/0.9.0.M2/aether-util-0.9.0.M2.jar
[ERROR] urls[23] = file:/home/msjimlau/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[ERROR] urls[24] = file:/home/msjimlau/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[ERROR] urls[25] = file:/home/msjimlau/.m2/repository/org/sonatype/sisu/sisu-guice/3.1.0/sisu-guice-3.1.0-no_aop.jar
[ERROR] urls[26] = file:/home/msjimlau/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[ERROR] urls[27] = file:/home/msjimlau/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.0.0.M5/org.eclipse.sisu.inject-0.0.0.M5.jar
[ERROR] urls[28] = file:/home/msjimlau/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.19/plexus-interpolation-1.19.jar
[ERROR] urls[29] = file:/home/msjimlau/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[30] = file:/home/msjimlau/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[31] = file:/home/msjimlau/.m2/repository/org/codehaus/plexus/plexus-utils/3.3.0/plexus-utils-3.3.0.jar
[ERROR] urls[32] = file:/home/msjimlau/.m2/repository/org/apache/maven/enforcer/enforcer-api/3.0.0/enforcer-api-3.0.0.jar
[ERROR] urls[33] = file:/home/msjimlau/.m2/repository/org/apache/maven/enforcer/enforcer-rules/3.0.0/enforcer-rules-3.0.0.jar
[ERROR] urls[34] = file:/home/msjimlau/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.2.0/maven-common-artifact-filters-3.2.0.jar
[ERROR] urls[35] = file:/home/msjimlau/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.3.3/maven-shared-utils-3.3.3.jar
[ERROR] urls[36] = file:/home/msjimlau/.m2/repository/org/apache/commons/commons-lang3/3.12.0/commons-lang3-3.12.0.jar
[ERROR] urls[37] = file:/home/msjimlau/.m2/repository/commons-codec/commons-codec/1.15/commons-codec-1.15.jar
[ERROR] urls[38] = file:/home/msjimlau/.m2/repository/commons-io/commons-io/2.11.0/commons-io-2.11.0.jar
[ERROR] urls[39] = file:/home/msjimlau/.m2/repository/org/apache-extras/beanshell/bsh/2.0b6/bsh-2.0b6.jar
[ERROR] urls[40] = file:/home/msjimlau/.m2/repository/org/apache/maven/resolver/maven-resolver-util/1.6.1/maven-resolver-util-1.6.1.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
```

--- a/README.md
+++ b/README.md
@@ -31,7 +31,7 @@ For information on how to use the integrations see the project documentation:
 ### Requirements
 
 * [Apache Maven](https://maven.apache.org/) 3.3+ (prefer to use included `mvnw`)
-* JDK 7+ (10 is **NOT** supported)
+* JDK 8+ (10 is **NOT** supported)
 
 ### Build
 

--- a/common/pom.xml
+++ b/common/pom.xml
@@ -46,7 +46,6 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-artifact</artifactId>
-      <version>3.0</version>
     </dependency>
 
     <dependency>

--- a/common/src/main/java/org/sonatype/ossindex/maven/common/ComponentReportAssistant.java
+++ b/common/src/main/java/org/sonatype/ossindex/maven/common/ComponentReportAssistant.java
@@ -124,7 +124,7 @@ public class ComponentReportAssistant
    */
   @VisibleForTesting
   static PackageUrl packageUrl(final Artifact artifact) {
-    return new PackageUrl.Builder()
+    return PackageUrl.builder()
         .type("maven")
         .namespace(artifact.getGroupId())
         .name(artifact.getArtifactId())

--- a/enforcer-rules/pom.xml
+++ b/enforcer-rules/pom.xml
@@ -41,7 +41,7 @@
     <dependency>
       <groupId>org.apache.maven.enforcer</groupId>
       <artifactId>enforcer-api</artifactId>
-      <version>3.0.0-M1</version>
+      <version>3.0.0</version>
     </dependency>
 
     <dependency>
@@ -64,11 +64,9 @@
       <artifactId>maven-artifact</artifactId>
     </dependency>
 
-    <!-- maven-enforcer-plugin uses older maven-dependency-tree 2.x API and is not compatible with latest 3.x -->
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-dependency-tree</artifactId>
-      <version>2.2</version>
     </dependency>
 
     <dependency>

--- a/enforcer-rules/pom.xml
+++ b/enforcer-rules/pom.xml
@@ -27,11 +27,6 @@
   <name>${project.groupId}:${project.artifactId}</name>
   <packaging>jar</packaging>
 
-  <properties>
-    <!-- maven-enforcer-plugin relies on older maven implementations -->
-    <apache-maven.version>3.0</apache-maven.version>
-  </properties>
-
   <dependencies>
     <dependency>
       <groupId>org.sonatype.ossindex.maven</groupId>

--- a/enforcer-rules/src/main/java/org/sonatype/ossindex/maven/enforcer/BanVulnerableDependencies.java
+++ b/enforcer-rules/src/main/java/org/sonatype/ossindex/maven/enforcer/BanVulnerableDependencies.java
@@ -20,6 +20,8 @@ import java.util.Set;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
+import org.apache.maven.project.DefaultProjectBuildingRequest;
+import org.apache.maven.project.ProjectBuildingRequest;
 import org.sonatype.ossindex.maven.common.ComponentReportAssistant;
 import org.sonatype.ossindex.maven.common.ComponentReportRequest;
 import org.sonatype.ossindex.maven.common.ComponentReportResult;
@@ -312,7 +314,11 @@ public class BanVulnerableDependencies
         artifactFilter = new CumulativeScopeArtifactFilter(scopes);
       }
 
-      DependencyNode node = graphBuilder.buildDependencyGraph(project, artifactFilter);
+      ProjectBuildingRequest buildingRequest =
+              new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+      buildingRequest.setProject( project );
+
+      DependencyNode node = graphBuilder.buildDependencyGraph(buildingRequest, artifactFilter);
       collectArtifacts(result, node);
 
       return result;

--- a/pom.xml
+++ b/pom.xml
@@ -95,7 +95,8 @@
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
 
-    <apache-maven.version>3.0</apache-maven.version>
+    <apache-maven.version>3.5.0</apache-maven.version>
+    <maven.plugin-annotations.version>3.6.0</maven.plugin-annotations.version>
     <jackson.version>2.9.10</jackson.version>
     <dionysus.version>1.0.3</dionysus.version>
 
@@ -137,7 +138,7 @@
       <dependency>
         <groupId>org.sonatype.ossindex</groupId>
         <artifactId>ossindex-service-client</artifactId>
-        <version>1.1.1</version>
+        <version>1.8.1</version>
       </dependency>
 
       <dependency>
@@ -149,7 +150,7 @@
       <dependency>
         <groupId>org.apache.maven.plugin-tools</groupId>
         <artifactId>maven-plugin-annotations</artifactId>
-        <version>${apache-maven.version}</version>
+        <version>${maven.plugin-annotations.version}</version>
       </dependency>
 
       <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -92,8 +92,8 @@
   </distributionManagement>
 
   <properties>
-    <maven.compiler.source>1.7</maven.compiler.source>
-    <maven.compiler.target>1.7</maven.compiler.target>
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <maven.compiler.target>1.8</maven.compiler.target>
 
     <apache-maven.version>3.0</apache-maven.version>
     <jackson.version>2.9.10</jackson.version>
@@ -185,7 +185,7 @@
       <dependency>
         <groupId>org.apache.maven.shared</groupId>
         <artifactId>maven-dependency-tree</artifactId>
-        <version>3.0.1</version>
+        <version>3.1.0</version>
       </dependency>
 
       <dependency>
@@ -365,7 +365,7 @@
             <configuration>
               <rules>
                 <byteCodeRule implementation="org.owasp.maven.enforcer.rule.ClassFileFormatRule">
-                  <supportedClassFileFormat>51</supportedClassFileFormat>
+                  <supportedClassFileFormat>52</supportedClassFileFormat>
                 </byteCodeRule>
               </rules>
             </configuration>
@@ -379,7 +379,7 @@
         <configuration>
           <signature>
             <groupId>org.codehaus.mojo.signature</groupId>
-            <artifactId>java17</artifactId>
+            <artifactId>java18</artifactId>
             <version>1.0</version>
           </signature>
         </configuration>

